actions/sync_gws_mailing_lists.py

```diff
@@ -32,62 +35,116 @@
 logger = logging.getLogger('sync_gws_mailing_lists')
 
 
-def get_all_group_members(group_email, gws_members_client):
-    """Get email addresses of all members of a Google Workspace members"""
+class GroupDoesNotExist(Exception):
+    pass
+
+
+def get_gws_group_members(group_email, gws_members_client) -> list:
+    """Return a list of Google Workspace group member dicts.
+
+    Specification of group member dictionary is here:
+    https://developers.google.com/admin-sdk/directory/reference/rest/v1/members#Member
+
+    Args:
+        group_email (str): Google Workspace group email
+        gws_members_client (googleapiclient.discovery.Resource): Admin API Members resource
+
+    Returns:
+        list: list of member dicts
+    """
+    ret = []
     req = gws_members_client.list(groupKey=group_email)
-    members = []
     while req is not None:
         res = req.execute()
         if 'members' in res:
-            members.extend(m['email'] for m in res['members'])
-        else:
-            break
+            ret.extend(res['members'])
         req = gws_members_client.list_next(req, res)
-    return members
+    return ret
 
 
-async def sync_gws_mailing_lists(gws_members_client, keycloak_client, dryrun=False):
+async def get_kc_group_canonical_emails(group_path, keycloak_client):
+    """Return a list of canonical emails of members of a KeyCloak group."""
+    try:
+        usernames = await get_group_membership(group_path, rest_client=keycloak_client)
+    except Exception as e:
+        if 'does not exist' in e.args[0]:
+            raise GroupDoesNotExist
+        else:
+            raise
+    users = [await user_info(u, rest_client=keycloak_client) for u in usernames]
+    emails = [u['attributes']['canonical_email'] for u in users]
+    return emails
+
+
+def sync_gws_group_role(group_email, role, role_emails, gws_members_client, dryrun):
+    """Sync Google Workspace group members of the given role to the given email list """
+    group_members = get_gws_group_members(group_email, gws_members_client)
+    initial_this_role_emails = [m['email'] for m in group_members if m['role'] == role]
+    initial_other_roles_emails = [m['email'] for m in group_members if m['role'] != role]
```

Comment:
Consider for these two lines using the set comprehension, if we're only going to use them as sets.
```suggestion
    initial_other_roles_emails = {m['email'] for m in group_members if m['role'] != role}
```

Reply:
I very much appreciate comments on style!